WordPress/gutenberg
# [edit-widgets] Register widgets as block variations

# Description

This PR ships first working version of widgets as block variations. The root idea is to use inversion of control to register the `legacy-widget` block - this could be useful with other blocks as well, let's consider adding core support for that.

<img width="355" alt="Zrzut ekranu 2020-08-28 o 20 32 42" src="https://user-images.githubusercontent.com/205419/91606135-d6a5fc80-e971-11ea-8df4-2c50e10312c2.png">

## How has this been tested?

1. Apply this branch
1. Apply the following diff locally: https://gist.github.com/adamziel/31afaca71c522abec5918e58d748ddf2 
1. Go to the widgets screen, open local inserter (the global one won't work until #24822 or it's subset lands).
1. Confirm you're able to find some widgets visible on the screenshot above, add RSS widget and confirm the form shows up correctly. 
1. Search for marquee widget and confirm it also works as expected.
1. Test saving and refreshing.

## Other considerations

With this PR it's possible to create multiple reference widgets. Unfortunately `supports.multiple=false` only works for the block, not for it's variations. Let's get this PR merged regardless and explore that issue separately.
